plugin/trino-hive-hadoop2/src/test/java/io/trino/plugin/hive/TestHivePlugin.java

```diff
@@ -126,6 +128,55 @@ public void testGlueMetastore()
                 .hasMessageContaining("Error: Configuration property 'hive.metastore.uri' was not used");
     }
 
+    @Test
+    public void testGlueMetastoreCustomAwsCredentialsProvider()
+            throws IOException
+    {
+        ConnectorFactory factory = getHiveConnectorFactory();
+        Path customCredConf = Files.createTempFile(null, null);
+
+        assertThatThrownBy(() -> factory.create(
+                "test",
+                ImmutableMap.of(
+                        "hive.metastore", "glue",
+                        "hive.metastore.glue.region", "us-east-2",
+                        "hive.metastore.glue.aws-credentials-providerconf", customCredConf.toString()),
+                new TestingConnectorContext()))
+                .hasMessageContaining("AwsCredentialsProvider class must be set when AwsCredentialsProviderConf is set");
+
+        factory.create(
+                        "test",
+                        ImmutableMap.of(
+                                "hive.metastore", "glue",
+                                "hive.metastore.glue.region", "us-east-2",
+                                "hive.metastore.glue.aws-credentials-provider", TestCustomAwsProvider.class.getName(),
+                                "hive.metastore.glue.aws-credentials-providerconf", customCredConf.toString()),
+                        new TestingConnectorContext())
+                .shutdown();
+    }
+
+    public static class TestCustomAwsProvider
+            implements AWSCredentialsProvider
+    {
+
+        public TestCustomAwsProvider(File configFile)
+        {
+
+        }
```

Comment:
```suggestion
        public TestCustomAwsProvider(File configFile) {}
```

and for `refresh` method too.